conans/client/tools/oss.py

```diff
@@ -48,16 +48,17 @@ def detected_os():
 def detected_architecture():
     # FIXME: Very weak check but not very common to run conan in other architectures
     machine = platform.machine()
+    os_info = OSInfo()
```

Comment:
if methods are static, we don't need an instance, right?

Reply:
Some members like `is_solaris` are initialized in the ctor; so as it is implemented right now, we need an instance

Reply:
oh I see now